convert_checkpoints.py

```diff
@@ -268,6 +275,84 @@ def _load_from_local(input_ckpt_dir: epath.Path):
   return checkpoints, params
 
 
+def _load_hf_llama_weight(input_ckpt_dir: epath.Path):
+  print(f"Loading checkpoint files from {input_ckpt_dir}.")
+  safetensors_files = input_ckpt_dir.glob("*.safetensors")
+  if len(list(safetensors_files)) == 0:
+    raise ValueError(
+        f"No *.safetensors found in the input dir {input_ckpt_dir}"
+    )
+  checkpoint = {}
+  for st_f in safetensors_files:
+    with safe_open(st_f, framework="pt", device="cpu") as f:
+      for key in f.keys():
+        if "inv_freq" in key:
+          # Don't include 'rotary_emb.inv_freq' in the converted
+          # checkpoint, because in JetStream implementation we
+          # precompute it during weight loading.
+          continue
+        new_key = key
+        # Remove 'model.' prefix for all weights.
+        prefix_to_remove = "model."
+        if key.startswith(prefix_to_remove):
+          new_key = new_key.removeprefix(prefix_to_remove)
+
+        # Weight name substring mapping between hf and jetstream.
+        _load_hf_llama_weight.hf_to_jetstream_keys_mapping = {
+            "lm_head": "output",
+            "embed_tokens": "tok_embeddings",
+            "input_layernorm": "attention_norm",
+            "post_attention_layernorm": "ffn_norm",
+            "self_attn.q_proj": "attention.wq",
+            "self_attn.k_proj": "attention.wk",
+            "self_attn.v_proj": "attention.wv",
+            "self_attn.o_proj": "attention.wo",
+            "mlp.gate_proj": "feed_forward.w1",
```

Comment:
I feel [gate|down|up]_proj are more read friendly than w1, w2 and w3.  @qihqi Shall we consider rename it to proj related name in default checkpoint convert?

Reply:
Yeah, this makes sense. Also want to note that original llama weight is using w1/2/3 https://github.com/meta-llama/llama3/blob/main/llama/model.py#L219. If we change it we need to do the name mapping for the original llama weight.